src/frontend/src/handler/mod.rs

```diff
@@ -467,17 +471,18 @@ pub async fn handle(
         Statement::AlterSystem { param, value } => {
             alter_system::handle_alter_system(handler_args, param, value).await
         }
-        // Ignore `StartTransaction` and `BEGIN`,`Abort`,`Rollback`,`Commit`temporarily.Its not
-        // final implementation.
+        // Ignore `Abort`,`Rollback`, etc. temporarily. It's not the final implementation.
         // 1. Fully support transaction is too hard and gives few benefits to us.
         // 2. Some client e.g. psycopg2 will use this statement.
         // TODO: Tracking issues #2595 #2541
-        Statement::StartTransaction { .. } => Ok(PgResponse::builder(START_TRANSACTION)
-            .notice(IGNORE_NOTICE)
-            .into()),
-        Statement::BEGIN { .. } => Ok(PgResponse::builder(BEGIN).notice(IGNORE_NOTICE).into()),
-        Statement::Abort { .. } => Ok(PgResponse::builder(ABORT).notice(IGNORE_NOTICE).into()),
-        Statement::Commit { .. } => Ok(PgResponse::builder(COMMIT).notice(IGNORE_NOTICE).into()),
+        Statement::StartTransaction { modes } => {
+            transaction::handle_begin(handler_args, START_TRANSACTION, modes).await
+        }
+        Statement::Begin { modes } => transaction::handle_begin(handler_args, BEGIN, modes).await,
+        Statement::Commit { chain } => {
+            transaction::handle_commit(handler_args, COMMIT, chain).await
+        }
+        Statement::Abort => Ok(PgResponse::builder(ABORT).notice(IGNORE_NOTICE).into()),
```

Comment:
`Abort` and `rollback` should terminate a transaction just like `commit`.

Reply:
Indeed, otherwise, even after users abort a read-only transaction, they always have the same view as the prior transaction.

Reply:
Fixed. Since we only have read-only transaction now, its behavior is exactly the same as `COMMIT` that does nothing.